### PR TITLE
Add 7.10 and 8.0 to test-matrix

### PR DESCRIPTION
**Pull request by hvr**
...and update to use cabal 1.24

### CODE REVIEW

**Comment by hvr** [comment]
btw, this PR subsumes #181

File: .travis.yml [hunk 1]
```diff
@@ -13,15 +15,8 @@ matrix:
...
- - export PATH=/opt/ghc/$GHCVER/bin:/opt/cabal/1.18/bin:$HOME/.cabal/bin:$PATH
- - |
-   if [ $GHCVER = "head" ] || [ ${GHCVER%.*} = "7.8" ]; then
-     travis_retry sudo apt-get install happy-1.19.4 alex-3.1.3
-     export PATH=/opt/alex/3.1.3/bin:/opt/happy/1.19.4/bin:$PATH
-   else
-     travis_retry sudo apt-get install happy alex
-   fi
+ - travis_retry sudo apt-get install cabal-install-1.24 ghc-$GHCVER happy-1.19.5 alex-3.1.7
+ - export PATH=/opt/alex/3.1.7/bin:/opt/happy/1.19.5/bin:/opt/ghc/$GHCVER/bin:/opt/cabal/1.24/bin:$HOME/.cabal/bin:$PATH
```
Comment (HuwCampbell) [inline .travis.yml]: What was your reasoning for this change?

File: .travis.yml [hunk 2]
```diff
@@ -35,4 +30,4 @@ script:
    if [ $GHCVER = "head" ] || [ ${GHCVER%.*} = "7.8" ] || [ ${GHCVER%.*} = "7.6" ] || [ ${GHCVER%.*} = "7.4" ] ; then
       cabal install --constraint="optparse-applicative < $VERSION"
       optparse-applicative-tests
-   fi
\ No newline at end of file
+   fi
```
Comment (HuwCampbell) [inline .travis.yml]: Please ensure that the tests run as well (2 lines up).

File: .travis.yml (hunk 1 above)
Comment (hvr) [inline .travis.yml]: tbh, I wasn't sure why the original travis script tried to use the stock `alex`/`happy` packages, so I just simplified that part to use the same versions for all matrix entries

File: .travis.yml (hunk 2 above)
Comment (hvr) [inline .travis.yml]: Sadly wouldn't work yet for GHC 8.0, as `tasty` has no install-plan for GHC 8 yet....

**Comment by HuwCampbell** [comment]
Thanks very much

File: .travis.yml (hunk 2 above)
Comment (HuwCampbell) [inline .travis.yml]: I've added 7.10 in #185

**Pull request merged**